src/paragraph/Paragraph.tsx

```diff
@@ -7,59 +7,68 @@ import { StyledParagraph } from './Paragraph.style'
+type ParagraphBaseProps = Readonly<{
   className?: string
-  children: string
-  isExpandable?: boolean
-  expandLabel?: string
   itemProp?: string
 }>
 
-type ParagraphState = {
-  isExpanded: boolean
+export type ParagraphSimpleProps = ParagraphBaseProps &
+  Readonly<{
+    children: React.ReactNode
+  }>
+
+export type ParagraphExpandableProps = ParagraphBaseProps &
+  Readonly<{
+    isExpandable: true
+    expandLabel: string
+    children: string
+  }>
+
+export type ParagraphProps = ParagraphSimpleProps | ParagraphExpandableProps
+
+const truncateText = (text: string) => `${text.substring(0, DEFAULT_MAX_CHAR_SIZE)}…`
+
+function SimpleParagraph(props: ParagraphSimpleProps): JSX.Element {
+  const { className, itemProp, children } = props
+
+  return (
+    <StyledParagraph className={cc(className)} role="presentation">
+      <Text itemProp={itemProp} tag={TextTagType.PARAGRAPH} newlineToBr>
+        {children}
+      </Text>
+    </StyledParagraph>
+  )
 }
 
-export class Paragraph extends PureComponent<ParagraphProps> {
-  static defaultProps: Partial<ParagraphProps> = {
-    className: '',
-    isExpandable: false,
-    expandLabel: '',
-    itemProp: null,
-  }
+function ExpandableParagraph(props: ParagraphExpandableProps): JSX.Element {
+  const { className, itemProp, expandLabel, children: originalContent } = props
 
-  state: ParagraphState = {
-    isExpanded: false,
-  }
+  const [isExpanded, setIsExpanded] = useState(originalContent.length < DEFAULT_MAX_CHAR_SIZE)
 
-  expandParagraph = () => {
-    this.setState({ isExpanded: true })
+  const expandParagraph = () => {
+    setIsExpanded(true)
   }
 
-  getTruncatedText() {
-    return `${this.props.children.substring(0, DEFAULT_MAX_CHAR_SIZE)}…`
-  }
+  const content = isExpanded ? originalContent : truncateText(originalContent)
 
-  render() {
-    const { className, children: originalContent, isExpandable, expandLabel, itemProp } = this.props
-    const { isExpanded } = this.state
-
-    const isContentTruncated =
-      isExpandable && !isExpanded && originalContent.length > DEFAULT_MAX_CHAR_SIZE
-    const content = isContentTruncated ? this.getTruncatedText() : originalContent
-
-    const readMoreButton = isContentTruncated && (
-      <Button status={ButtonStatus.UNSTYLED} onClick={this.expandParagraph} className="mt-s">
-        {expandLabel}
-      </Button>
-    )
-
-    return (
-      <StyledParagraph className={cc(className)} role="presentation">
-        <Text {...(itemProp && { itemProp })} tag={TextTagType.PARAGRAPH} newlineToBr>
-          {content}
-        </Text>
-        {readMoreButton}
-      </StyledParagraph>
-    )
+  return (
+    <StyledParagraph className={cc(className)} role="presentation">
+      <Text itemProp={itemProp} tag={TextTagType.PARAGRAPH} newlineToBr>
+        {content}
+      </Text>
+      {!isExpanded && (
+        <Button status={ButtonStatus.UNSTYLED} onClick={expandParagraph} className="mt-s">
```

Comment:
We should avoid these kind of classname here. Can you remove this `mt-s` and add the margin in `Paragraph.style.tsx`

Reply:
Missing some ARIA controls. Please check with @nickoola  to help you out

Reply:
@gabrielagabriel you right, there is also the focus to manage after click on the expand button.
But the a11y issue here is that we hide the button after click. 
Before adding some stuff for a11y we should ask Pixar (Jonas) if we can update the specs to keep the expand button visible. Like for the `HighlightSection`.